src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -39,7 +39,11 @@ ClusterTaskManager::ClusterTaskManager(
     std::function<bool(const std::vector<ObjectID> &object_ids,
                        std::vector<std::unique_ptr<RayObject>> *results)>
         get_task_arguments,
-    size_t max_pinned_task_arguments_bytes)
+    size_t max_pinned_task_arguments_bytes,
+    std::function<std::shared_ptr<boost::asio::deadline_timer>(std::function<void()>,
+                                                               int64_t)>
+        execute_after,
+    std::function<int64_t(void)> get_time, int64_t sched_cls_cap_interval_ms)
```

Comment:
I actually don't like get_time method here since it's purely for testing and kind of making things more complicated (one more parameters).

I'm ok with this right now, but we probably can do a better job later.

I'm thinking we probably can have a lib:

```
#ifdef UNIT_TEST
int64_t time = 0;
auto GetCurrentTimeNano = []() {
    return time;
}
#else 
int64_t GetCurrentTimeNano() {
    return absl::GetCurrentTImeNano();
}
#endif
```